docs/Test.md

```diff
@@ -0,0 +1,154 @@
+
+- Date and time should be mocked/set to a known value
+- Tests should cause no side effects - they should create and delete any
+  resources they need in the scope of that test/test suite
+- Tests (regardless of style) should be structured in the following way:
+  1. Perform any required setup
+  2. Perform an action (or actions)
+  3. Assert the result
+  4. Perform any required clean up
+
+## Code coverage
+
+In this repo, we aim to have 100% code coverage across all components. This code
+coverage should be achieved through useful functional/behavioural tests and
+not just units to increase the coverage. We want all lines to be verified
+through meaninful tests.
+
+## Accessibility testing
+
+To be discussed at https://github.com/strimzi/strimzi-ui/issues/20
+
+## Test tooling used
+
+The current set of tooling used to implement tests can be found below. In
+addition to this set, a library of [test utilities exist](../src/TestUtils/README.md)
+to help enable the fast and reliable creation and maintaince of tests going
+forwards.
+
+The current testing tooling used for UI code is as follows:
+
+- [Jest](https://jestjs.io/) - as a test runner. Used to run functional and behavioural tests
+- [RTL (react-test-library)](https://testing-library.com/docs/react-testing-library/intro) - used to emulate a browser DOM and mimic
+  user interaction with React components
+- [Cypress](https://www.cypress.io/) - as an end-to-end test runner
+- [cypress-cucumber-preprocessor](https://github.com/TheBrainFamily/cypress-cucumber-preprocessor) - as a step definition library
+
+## Test utilities
+
+Where possible, single responsibility test utilities should be created
+to abstract common test logic so that it can be easily re-used and lifts
+the barrier to entry to write new tests. These utilities should have simple
+function signatures that are well documented. Once implemented these should
+not be changed so that function complexity does not grow. If additional behaviour
+is required later on, these should be extended through new utility functions.
+
+## Testing contexts (an example of using test utilities)
+
+Testing contexts is an example of having common test logic that can be
+abstracted away. Here, the function `renderWithContextProviders` has been
+provided that extends the base rtl `render` function but can be given a list
+of context providers and their values that will wrap the provided children.
+Using this, a common render can be defined in a test file for contexts so that
+each test only needs to worry about using a consumer. For example:
+
+```
+render = (children) =>
+  renderWithContextProviders(children, {}, [
+    { provider: MyProvicer1, value: providerValue1 },
+    { provider: MyProvider2, value: providerValue2 },
+  ]);
+```
+
+This can be defined in a test file so that tests can still use a `render`
+function in the same way they usually would with rtl but these renders will be
+wrapped in the providers defined in the above function.
```

Comment:
Would suggest linking off to the `utils/test/README` here, and having links in that README to specific helpers (like the one here for contexts). I can see this list growing, making this document harder to follow.